[3.10] Compatibilitycheck

GeraintEdwards opened the pull request:
Pull Request for Issue # .

### Summary of Changes

The current upgrade compatibility checks for extensions is not easy to understand, the results are not grouped together in any meaningful way, they also do not provide a comparison between available compatible versions for currently installed version of Joomla and the targetted upgrade version.

### Testing Instructions

1. Apply patch to Joomla 3.10-dev branch
2. Set the upgrade server settings for Joomla to target one of the 4.x releases e.g. https://update.joomla.org/core/nightlies/next_major_list.xml
3. check for new Joomla upgrades

For convenience only I have provided a set of 'dummy' extensions to illustrate the different sort of results you could achieve - please just install them before checking for new Joomla upgrades.

[testuptodatepackage.zip](https://github.com/joomla/joomla-cms/files/4751464/testuptodatepackage.zip)
[testwuj3ej4package.zip](https://github.com/joomla/joomla-cms/files/4751465/testwuj3ej4package.zip)
[testuxpackage.zip](https://github.com/joomla/joomla-cms/files/4751466/testuxpackage.zip)
[testwuj3n4package.zip](https://github.com/joomla/joomla-cms/files/4751467/testwuj3n4package.zip)
[testwuvcpackage.zip](https://github.com/joomla/joomla-cms/files/4751468/testwuvcpackage.zip)
[testxupackage.zip](https://github.com/joomla/joomla-cms/files/4751469/testxupackage.zip)

The results will look like this screenshot.

![Screenshot_2020-06-09 Joomla Update - Joomla 3 10 Git - Administration](https://user-images.githubusercontent.com/839810/84135378-d0bf1000-aa41-11ea-996c-c37af6b1bdf9.png)

### Documentation Changes Required
Pre-update check docs will need amending

Geraint2 committed:
Alternative presentation of compatibility checks for Joomla 4 from Joomla 3.10

Geraint2 committed:
Alternative presentation of compatibility checks for Joomla 4 from Joomla 3.10

Geraint2 committed:
remove accidental commit of .gitignore

Geraint2 committed:
Handle situation where we are running 3.10 with an extension what doesn't say its compatible with 3.10 - was incorrectly being reported as not supporting Joomla updater

Geraint2 committed:
Adding user guidance notes for update check

Geraint2 committed:
Merge remote-tracking branch 'remotes/upstream/3.10-dev' into compatibilitycheck

richard67 commented:
@GeraintEdwards Please fix code style errors as reported by drone [https://ci.joomla.org/joomla/joomla-cms/32942/1/2](https://ci.joomla.org/joomla/joomla-cms/32942/1/2)

toivo commented:
I have tested this item :white_check_mark: successfully on 19135972d18876bb0b00527f5cd890a9a7f706fe<br /><br />Tested successfully with 3.10-dev from Nightly Build 9 May.

Test environment: Wampserver 3.2.2  MySQL 8.0.19  PHP 7.4.5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29524">issues.joomla.org/tracker/joomla-cms/29524</a>.</sub>

richard67 commented:
@toivo How can you give a good test result if code style tests (PHPCS) fail in drone?

Quy commented:
Please add width style to `<th>` so the columns align.

brianteeman commented:
@richard67 the drone tests dont show up in the issue tracker

brianteeman commented:
Could you please try to write shorter messages. They are very hard to understand.

Geraint2 committed:
code style

GeraintEdwards commented:
@brianteeman Do you mean the explanatory comments under the headings should be shorter?  

The idea is to put enough detail to explain what the category of results mean so that there is less reliance on documentation.  Its important that users understand the importance of this output otherwise we may get users performing bad upgrades and Joomla being blamed.

I'm very open to different wording or direct links to relevant documentation.  

Any specific ideas?

brianteeman commented:
Research has proven that the more text you write then the less people read.

GeraintEdwards commented:
@brianteeman I can make a couple of the messages 'context sensitive' e.g. only show them if the relevant situation arises e.g. Joomla 3.x update available that isn't available in Joomla 4.x .  

@Quy I agree about aligning the column widths (subject to maintaining some sort of good behaviour on different sized screens).

What I'm most interested in. initially. is people's reaction to this alternative presentation of the information and if its is likely to help users understand the upgrade process better and reduce the risks of broken sites when Joomla is upgraded.  If its seen as a good idea/approach then it can be polished up a bit more.??

bayareajenn commented:
I have been testing this for a couple of weeks. I like how it's organised by tag and the colour of the bars. It makes more sense to the user than the mixed up ordering of the tags in the existing version of pre-update check.

Couple of typos - the same for each description under the red bars. "if" should be "is." Maybe it doesn't matter though...

Thoughts:

1. The second red bar I think should be gray - same as the missing compatibility tag is in the current version of pre-update check since a number of them will migrate on the one click to 4. To me red means NO and some of these are definitely YES.

2. The titles in the bars I think should be clearer and ditch the description below. Suggestions:

Green - Yes, tagged as compatible with Joomla 4.
Yellow/Orange - May need an update to be compatible. Update or check with developer.
Red - No compatiblity with Joomla 4. Check with developer on migration path.
Gray/Second Red - Missing compatibility tag. Check with developer on migration path.

3. We can put the deeper descriptions and more explanation in the doc (docs.joomla.org/Pre-Update_Check). Note that is a work in progress at this moment.

4. Add to the Notice at the bottom:

Pre-Update Check Documentation (link to docs.joomla.org/Pre-Update_Check)
Dev Resources: Deploying an Update Server (https://docs.joomla.org/Deploying_an_Update_Server), Joomla Update System Requirement (https://extensions.joomla.org/support/knowledgebase/submission-requirements/joomla-update-system-requirement/), and Extension Manager Update (https://docs.joomla.org/Help39:Extensions_Extension_Manager_Update)

I will wait to test until the issue Richard is talking about is resolved. However I have been testing for some time. I think this version will make more sense to more users and be easier to understand - with a few changes for more clarity.

richard67 commented:
@bayareajenn Code style issue is solved. Or did you mean something else?

bayareajenn commented:
> 
> 
> @bayareajenn Code style issue is solved. Or did you mean something else?

No, that's what I meant. I don't really know how to read github very well. Getting better. So ok. I'll test. Thanks Richard.

toivo commented:
@richard67 I did not check the code in diff other than check that it did not have any SQL statements.

@GeraintEdwards The Extensions Pre-Update Check report looks good and I strongly believe it will assist users and get them to update or uninstall out-of-date extensions before upgrading Joomla.  It will make the task of volunteers in the Joomla! Forum easier by spelling out extension by extension if they are compatible.    Any chance of getting at least a summary report in BBCode, similar to what we get from the Forum Post Assitant (FPA)?

Note to self: study the documentation and make one of my own 3.x test extensions upgradeable.

GeraintEdwards commented:
Is it worth me spending more time on this?  

Do people feel this approach adds value?

In which case I would make some of the messages context sensitive (i.e. only display relevant warnings) and align the columns in the tables.

Quy commented on administrator/components/com_joomlaupdate/controllers/update.php:
```diff
@@ -501,10 +501,51 @@ public function fetchExtensionCompatibility()
+		$upgradeCompatibilityStatus = $model->fetchCompatibility($extensionID, $joomlaTargetVersion);
+		$currentCompatibilityStatus = $model->fetchCompatibility($extensionID, $joomlaCurrentVersion);
+
+		$resultGroup = -1;
+		$upgradeWarning = 0;
+		if ($upgradeCompatibilityStatus->state == 1 && $upgradeCompatibilityStatus->compatibleVersion !== false)
```
```suggestion

		if ($upgradeCompatibilityStatus->state == 1 && $upgradeCompatibilityStatus->compatibleVersion !== false)
```

Quy replied:
```suggestion
		$upgradeWarning = 0;
	
		if ($upgradeCompatibilityStatus->state == 1 && $upgradeCompatibilityStatus->compatibleVersion !== false)
```

Quy commented on administrator/components/com_joomlaupdate/controllers/update.php:
```diff
@@ -501,10 +501,51 @@ public function fetchExtensionCompatibility()
+			'currentCompatibilityStatus' => $currentCompatibilityStatus,
+			'resultGroup' => $resultGroup,
+			'upgradeWarning' => $upgradeWarning
+			);
```
```suggestion
		);
```

alikon commented:
> Do people feel this approach adds value?

yes for me

GeraintEdwards committed:
Update administrator/components/com_joomlaupdate/controllers/update.php

Co-authored-by: Quy <quy@fluxbb.org>

GeraintEdwards committed:
Update administrator/components/com_joomlaupdate/controllers/update.php

Co-authored-by: Quy <quy@fluxbb.org>

Geraint2 committed:
Changed styling of compatibility feedback columns so that they are aligned
Make warning message display be conditional on being relevant to the displayed output.

GeraintEdwards commented:
@Quy I have now set column widths so that the columns all line up

@brianteeman I have make a couple of the warning messages context sensitive - i.e. they only appear when they are relevant.

richard67 commented on .gitignore:
```diff
@@ -230,3 +230,20 @@ composer.phar
 
 # Build related
 RoboFile.ini
+
+# JEvents
+**/com_jevents/**
+**/mod_jevents*/**
+**/googl/**
+**/jevmodal/**
+**/eventsearch/**
+**/gwejson/**
+**/jeventsinstaller/**
+**/jevents
+**/*_jevents*
+**/*_gwejson*
+**/*_eventsearch*
+**/robots.txt
+**/googl.xml
+
+.gitignore
```
Why these changes?

GeraintEdwards replied:
Whoops - will correct right away.  Should not have been checked with the commit

GeraintEdwards replied:
sorted

Geraint2 committed:
Revert accidentally committed file

Quy commented on administrator/components/com_joomlaupdate/views/default/tmpl/default.php:
```diff
@@ -18,10 +18,12 @@
+JText::script('COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSION_SERVER_ERROR');
 
 $latestJoomlaVersion = $this->updateInfo['latest'];
+$currentJoomlaVersion = $this->updateInfo['current'];
```
Go to Components > Joomla! Update without setting the upgrade server settings.

> PHP Notice:  Undefined index: current in \administrator\components\com_joomlaupdate\views\default\tmpl\default.php on line 26

GeraintEdwards replied:
Don't know why I didn't spot that - correcting now.

GeraintEdwards replied:
should be resolved now

Quy commented on administrator/language/en-GB/en-GB.com_joomlaupdate.ini:
```diff
@@ -46,14 +46,29 @@ COM_JOOMLAUPDATE_VIEW_DEFAULT_COMPATIBILITY_CHECK="Joomla! %s Compatibility Chec
 COM_JOOMLAUPDATE_VIEW_DEFAULT_DESCRIPTION_BREAK="Extensions marked with <span class='label label-important'>No</span> or <span class='label'>Missing Compatibility Tag</span> might break your website. Please consult with the developer before upgrading."
 COM_JOOMLAUPDATE_VIEW_DEFAULT_DESCRIPTION_MISSING_TAG="Extensions marked with <span class='label'>Missing Compatibility Tag</span> indicate the developer has not included <a href='https://docs.joomla.org/Special:MyLanguage/Deploying_an_Update_Server' target='_blank' rel='noopener noreferrer'>compatibility information.</a>"
 COM_JOOMLAUPDATE_VIEW_DEFAULT_DESCRIPTION_UPDATE_REQUIRED="Extensions marked with <span class='label label-warning'>Yes (X.X.X)</span> might require an update."
+COM_JOOMLAUPDATE_VIEW_DEFAULT_COMPATIBLE_UPGRADE_WARNING="Extensions marked with <span class='label label-warning'>X.X.X</span> have an extension upgrade available for the current version of Joomla which is not marked as compatible with the upgraded version of Joomla.  You should contact the extension developer for more information."
```
Please sort keys in alpha order.

GeraintEdwards replied:
should be resolved now

Quy commented on administrator/components/com_joomlaupdate/views/default/tmpl/default_preupdatecheck.php:
```diff
@@ -92,57 +92,88 @@
+								<th class="span2">
+									<?php echo JText::_('COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSION_TYPE'); ?>
+								</th>
+                                <th class="span2">
```
Please use tabs to indent.

GeraintEdwards replied:
should be resolved now

Geraint2 committed:
Use tabs for HTML spacing
Alphabetical language string key definition
Resolve undeclared updateInfo['current']

Quy commented on administrator/components/com_joomlaupdate/controllers/update.php:
```diff
@@ -501,10 +501,51 @@ public function fetchExtensionCompatibility()
+				$resultGroup = 1;
+			}
+
+			if ($currentCompatibilityStatus == 1)
```
> PHP Notice:  Object of class stdClass could not be converted to int on line 525

Geraint2 committed:
Missing ->state when checking "if ($currentCompatibilityStatus->state == 1)"

alikon commented:
@adj9 would you like to test this one ?

adj9 commented:
I have tested this item :white_check_mark: successfully on 6615f87fa2cb8a6eaecb32070592ea5f0615f992<br /><br />### Test phases:
- Installed all packages
- Upgrade from Custom URL to version 3.10
- Checking for updates
- Application of PR

### Results (3.9->4.0)
For minimal server requirements I have no problem, I use PHP and MySQL.
Pre-Check phases:
- Extensions Probably Compatible, passed 
- Extensions Requiring Upgrades to be Compatible, is Orange 
- Update Server Offers No Compatible Version, there is no compatibility
- Joomla Update System Not Supported, there is no compatibility 

I'm not continuing the migration to J! 4. 

As for the graphics, so far it is very minimal. I find it suitable to understand if the versions installed in the Joommla! environment are compatible with the migration version. With so many extensions installed it becomes more complicated because there is a lot of information.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29524">issues.joomla.org/tracker/joomla-cms/29524</a>.</sub>

alikon commented:
I have tested this item :white_check_mark: successfully on 6615f87fa2cb8a6eaecb32070592ea5f0615f992<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29524">issues.joomla.org/tracker/joomla-cms/29524</a>.</sub>

alikon commented:
rtc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29524">issues.joomla.org/tracker/joomla-cms/29524</a>.</sub>

GeraintEdwards committed:
Merge branch '3.10-dev' into compatibilitycheck

GeraintEdwards commented:
Should I make the equivalent changes as a PR for Joomla 4.0?

zero-24 commented:
> Should I make the equivalent changes as a PR for Joomla 4.0?

I have added this PR to my review list and I will get back to you soon. When merged this will be merged up to 4.x too so right now there is no 4.0 PR required.

> p.s. Looks like one of the tests failed and may just need rerunning.

Seems this is not realted to this PR.

GeraintEdwards commented:
> I have added this PR to my review list and I will get back to you soon. When merged this will be merged up to 4.x too so right now there is no 4.0 PR required.

It won't merge into Joomla 4.0 easily so I'm happy to help with changes required to get it working in J4.  I'll watch for you review - thanks

zero-24 commented:
Ok did round of tests here. Here are some issues i found as well as some suggestions

## Bugs

Seems I found an issue introduced here go to com_joomlaupdate -> try to navigate to "extensions -> manage -> install" -> works -> apply this patch see. It seems als sub menus are broken there: 
![image](https://user-images.githubusercontent.com/2596554/86526331-10d3b000-be93-11ea-920e-524abc6a0698.png)

---------------------------------------------------------------------------------------------------

Seems we also have an JS error when 0 extensions are installed:
![image](https://user-images.githubusercontent.com/2596554/86526351-70ca5680-be93-11ea-8346-8969187eeaf7.png)

## Suggestions

Install https://github.com/zero-24/plg_fields_privacycheckbox/releases/tag/1.0.4
This extension is only comatible with 3.x but there is a 4.x version all fine. The problem is that this is the only version installed but you still get this:
![image](https://user-images.githubusercontent.com/2596554/86526449-7d9b7a00-be94-11ea-8b1e-8eeb5144615f.png)

So the suggestion is to hide the sections where no extension is in so it does not send a confusing message on the first look ;)

---------------------------------------------------------------------------------------------------

I have just installed the demo extensions
![image](https://user-images.githubusercontent.com/2596554/86526515-2c3fba80-be95-11ea-86cd-58d254dd84ba.png)

Seems this message is dublicated now? I would suggest to remove the notice from the bottom.

---------------------------------------------------------------------------------------------------

I have a big screen (32") but still there is no end at the right side ;)
![image](https://user-images.githubusercontent.com/2596554/86526551-c69ffe00-be95-11ea-983e-8be3f37be4ff.png)

Can we add a bit padding as we have it on the left site to that table?

---------------------------------------------------------------------------------------------------

I will review and send some suggestions for the language keys too so we direct people and extension devs to dedicated docs.

Thanks for your work on this this already is a great step into the right direction @GeraintEdwards 👍

zero-24 commented on administrator/components/com_joomlaupdate/controllers/update.php:
```diff
@@ -501,10 +501,51 @@ public function fetchExtensionCompatibility()
 	{
 		$extensionID = $this->input->get('extension-id', '', 'DEFAULT');
 		$joomlaTargetVersion = $this->input->get('joomla-target-version', '', 'DEFAULT');
+		$joomlaCurrentVersion = $this->input->get('joomla-current-version', '', 'DEFAULT');
```
```suggestion
		$joomlaCurrentVersion = $this->input->get('joomla-current-version', '', JVERSION);
```

zero-24 commented on administrator/components/com_joomlaupdate/controllers/update.php:
```diff
@@ -501,10 +501,51 @@ public function fetchExtensionCompatibility()
+			'upgradeCompatibilityStatus' => $upgradeCompatibilityStatus,
+			'currentCompatibilityStatus' => $currentCompatibilityStatus,
+			'resultGroup' => $resultGroup,
+			'upgradeWarning' => $upgradeWarning
```
```suggestion
			'upgradeWarning' => $upgradeWarning,
```

zero-24 commented on administrator/components/com_joomlaupdate/views/default/tmpl/default_preupdatecheck.php:
```diff
@@ -92,57 +92,88 @@
+										<td class="extension-check span2"
+											data-extension-id="<?php echo $extension->extension_id; ?>"
+											data-extension-current-version="<?php echo $extension->version; ?>">
+											<img src="../media/system/images/mootree_loader.gif" />
```
```suggestion
											<img src="../media/jui/images/ajax-loader.gif" />
```

zero-24 replied:
Lets use the joomla ajax loader insted of the generic one ;)

zero-24 commented on administrator/language/en-GB/en-GB.com_joomlaupdate.ini:
```diff
@@ -43,17 +43,32 @@ COM_JOOMLAUPDATE_VIEW_COMPLETE_HEADING="Joomla Version Update Status"
 COM_JOOMLAUPDATE_VIEW_COMPLETE_MESSAGE="Your site has been updated. Your Joomla version is now %s."
 COM_JOOMLAUPDATE_VIEW_DEFAULT_ACTUAL="Actual"
 COM_JOOMLAUPDATE_VIEW_DEFAULT_COMPATIBILITY_CHECK="Joomla! %s Compatibility Check"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_COMPATIBLE_UPGRADE_WARNING="Extensions marked with <span class='label label-warning'>X.X.X</span> have an extension upgrade available for the current version of Joomla which is not marked as compatible with the upgraded version of Joomla.  You should contact the extension developer for more information."
```
```suggestion
COM_JOOMLAUPDATE_VIEW_DEFAULT_COMPATIBLE_UPGRADE_WARNING="Extensions marked with <span class='label label-warning'>X.X.X</span> have an extension upgrade available for the current version of Joomla which is not marked as compatible with the upgraded version of Joomla. You should contact the extension developer for more information."
```

GeraintEdwards replied:
When I learnt to type I was taught to always put a double space after a full stop and a single quote after a comma or semi-colon.  Old habits die hard!

zero-24 commented on administrator/language/en-GB/en-GB.com_joomlaupdate.ini:
```diff
@@ -43,17 +43,32 @@ COM_JOOMLAUPDATE_VIEW_COMPLETE_HEADING="Joomla Version Update Status"
 COM_JOOMLAUPDATE_VIEW_DEFAULT_DOWNLOAD_IN_PROGRESS="Downloading update file. Please wait ..."
 COM_JOOMLAUPDATE_VIEW_DEFAULT_FTP_DIRECTORY="FTP Root"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_RUNNING_PRE_UPDATE_CHECKS="Running Pre-Update Checks"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_RUNNING_PRE_UPDATE_CHECKS_NOTES="Please be patient whilst we run the pre-update checks on these extensions."
```
```suggestion
COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_RUNNING_PRE_UPDATE_CHECKS_NOTES="Please be patient whilst we run the pre-update checks on your extensions."
```

zero-24 commented on administrator/language/en-GB/en-GB.com_joomlaupdate.ini:
```diff
@@ -43,17 +43,32 @@ COM_JOOMLAUPDATE_VIEW_COMPLETE_HEADING="Joomla Version Update Status"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_RUNNING_PRE_UPDATE_CHECKS="Running Pre-Update Checks"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_RUNNING_PRE_UPDATE_CHECKS_NOTES="Please be patient whilst we run the pre-update checks on these extensions."
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_PROBABLY_COMPATIBLE="Extensions Probably Compatible"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_PROBABLY_COMPATIBLE_NOTES="<p>The extension developer has provided us with information to suggest that the currently installed version is compatible with the available upgraded version of Joomla.</p><p id='upgradeyellowwarning' class='hidden'>Please note that if you see a version highlighted as <span class='label label-warning'>n.n.n</span> then the extens ...
```
```suggestion
COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_PROBABLY_COMPATIBLE_NOTES="<p>The extension developer has provided us with information to suggest that the currently installed version is compatible with the available upgraded version of Joomla.</p><p id='upgradeyellowwarning' class='hidden'>Please note that if you see a version highlighted as <span class='label label-warning'>X.X.X</span> then the extension developer is offering a newer version of the extension for your current version of Joomla than they do for the new version of Joomla. You should check with the extension developer if this is correct before you ugprade Joomla.</p>"
```

zero-24 commented on administrator/language/en-GB/en-GB.com_joomlaupdate.ini:
```diff
@@ -43,17 +43,32 @@ COM_JOOMLAUPDATE_VIEW_COMPLETE_HEADING="Joomla Version Update Status"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_PROBABLY_COMPATIBLE_NOTES="<p>The extension developer has provided us with information to suggest that the currently installed version is compatible with the available upgraded version of Joomla.</p><p id='upgradeyellowwarning' class='hidden'>Please note that if you see a version highlighted as <span class='label label-warning'>n.n.n</span> then the extens ...
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_REQUIRING_UPGRADES_TO_BE_COMPATIBLE="Extensions Requiring Upgrades to be Compatible"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_REQUIRING_UPGRADES_TO_BE_COMPATIBLE_NOTES="<p>The extension developer is offering a new version of the extension that they say is compatible with the available upgraded version of Joomla.</p><p id='upgradeorangewarning' class='hidden'>Please take extra care if this upgraded version of the extension is not also listed as compatible with your current version ...
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_UPDATE_SERVER_OFFERS_NO_COMPATIBLE_VERSION="Update Server Offers No Compatible Version"
```
```suggestion
COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_UPDATE_SERVER_OFFERS_NO_COMPATIBLE_VERSION="Update Server offers no compatible version"
```

I'm not 100% sure on this one maybe @brianteeman can give advise on the capalization of the headings?

GeraintEdwards replied:
The other headings are long word capitalised  - so if we change this we should change the others.

zero-24 commented on administrator/language/en-GB/en-GB.com_joomlaupdate.ini:
```diff
@@ -43,17 +43,32 @@ COM_JOOMLAUPDATE_VIEW_COMPLETE_HEADING="Joomla Version Update Status"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_REQUIRING_UPGRADES_TO_BE_COMPATIBLE_NOTES="<p>The extension developer is offering a new version of the extension that they say is compatible with the available upgraded version of Joomla.</p><p id='upgradeorangewarning' class='hidden'>Please take extra care if this upgraded version of the extension is not also listed as compatible with your current version ...
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_UPDATE_SERVER_OFFERS_NO_COMPATIBLE_VERSION="Update Server Offers No Compatible Version"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_UPDATE_SERVER_OFFERS_NO_COMPATIBLE_VERSION_NOTES="The extension developer is not currently offering information about a version of the extension that is compatible with the available upgraded version of Joomla.  Please contact the extension developer for more information before attempting to upgrade Joomla."
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_JOOMLA_UPDATE_SYSTEM_NOT_SUPPORTED="Joomla Update System Not Supported"
```
```suggestion
COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_JOOMLA_UPDATE_SYSTEM_NOT_SUPPORTED="Joomla Update System is not supported"
```

zero-24 commented on administrator/language/en-GB/en-GB.com_joomlaupdate.ini:
```diff
@@ -43,17 +43,32 @@ COM_JOOMLAUPDATE_VIEW_COMPLETE_HEADING="Joomla Version Update Status"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_UPDATE_SERVER_OFFERS_NO_COMPATIBLE_VERSION="Update Server Offers No Compatible Version"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_UPDATE_SERVER_OFFERS_NO_COMPATIBLE_VERSION_NOTES="The extension developer is not currently offering information about a version of the extension that is compatible with the available upgraded version of Joomla.  Please contact the extension developer for more information before attempting to upgrade Joomla."
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_JOOMLA_UPDATE_SYSTEM_NOT_SUPPORTED="Joomla Update System Not Supported"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_JOOMLA_UPDATE_SYSTEM_NOT_SUPPORTED_NOTES="The extension developer is not currently offering any information about extension compatibility with different versions of Joomla.  Please contact the extension developer for more information before attempting to upgrade Joomla."
```
```suggestion
COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_JOOMLA_UPDATE_SYSTEM_NOT_SUPPORTED_NOTES="The extension developer is not currently offering <a href='https://docs.joomla.org/Special:MyLanguage/Deploying_an_Update_Server' target='_blank' rel='noopener noreferrer'>any information about extension compatibility with different versions of Joomla</a>. Please contact the extension developer for more information before attempting to upgrade Joomla."
```

zero-24 commented on administrator/templates/hathor/html/com_joomlaupdate/default/default.php:
```diff
@@ -18,10 +18,12 @@
 JText::script('COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSION_WARNING_UNKNOWN');
+JText::script('COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSION_SERVER_ERROR');
 
 $latestJoomlaVersion = $this->updateInfo['latest'];
```
```suggestion
$latestJoomlaVersion  = $this->updateInfo['latest'];
```

zero-24 commented on media/com_joomlaupdate/js/default.js:
```diff
@@ -49,6 +49,7 @@ function extractionMethodHandler(target, prefix)
     PreUpdateChecker.run = function () {
         // Get version of the available joomla update
         PreUpdateChecker.joomlaTargetVersion = window.joomlaTargetVersion;
+		PreUpdateChecker.joomlaCurrentVersion = window.joomlaCurrentVersion;
```
```suggestion
        PreUpdateChecker.joomlaCurrentVersion = window.joomlaCurrentVersion;
```

zero-24 commented on media/com_joomlaupdate/js/default.js:
```diff
@@ -71,19 +76,20 @@ function extractionMethodHandler(target, prefix)
-            state: PreUpdateChecker.STATE.SERVER_ERROR,
-            compatibleVersion: 0
+            compatibleVersion: 0,
+			serverError: 1
```
```suggestion
            serverError: 1
```

zero-24 commented on media/com_joomlaupdate/js/default.js:
```diff
@@ -57,6 +58,10 @@ function extractionMethodHandler(target, prefix)
             // function after completing the request
             PreUpdateChecker.checkCompatibility($(this), PreUpdateChecker.setResultView);
         });
+		document.getElementById().css('display', 'none')
```
```suggestion
        document.getElementById().css('display', 'none')
```

zero-24 commented on media/com_joomlaupdate/js/default.js:
```diff
@@ -57,6 +58,10 @@ function extractionMethodHandler(target, prefix)
             PreUpdateChecker.checkCompatibility($(this), PreUpdateChecker.setResultView);
         });
+		document.getElementById().css('display', 'none')
+		$('.compatibilitytypes').css('display', 'none').css('margin-left', 0);
```
```suggestion
        $('.compatibilitytypes').css('display', 'none').css('margin-left', 0);
```

zero-24 commented on media/com_joomlaupdate/js/default.js:
```diff
@@ -57,6 +58,10 @@ function extractionMethodHandler(target, prefix)
         });
+		document.getElementById().css('display', 'none')
+		$('.compatibilitytypes').css('display', 'none').css('margin-left', 0);
+		// The currently processing line should show until its finished
```
```suggestion
        // The currently processing line should show until its finished
```

zero-24 commented on media/com_joomlaupdate/js/default.js:
```diff
@@ -57,6 +58,10 @@ function extractionMethodHandler(target, prefix)
+		document.getElementById().css('display', 'none')
+		$('.compatibilitytypes').css('display', 'none').css('margin-left', 0);
+		// The currently processing line should show until its finished
+		$('#compatibilitytype0').css('display', 'block');
```
```suggestion
        $('#compatibilitytype0').css('display', 'block');
```

zero-24 commented on media/com_joomlaupdate/js/default.js:
```diff
@@ -71,19 +76,20 @@ function extractionMethodHandler(target, prefix)
         // Request the server to check the compatiblity for the passed extension and joomla version
         $.getJSON(PreUpdateChecker.config.serverUrl, {
             'joomla-target-version': PreUpdateChecker.joomlaTargetVersion,
+			'joomla-current-version': PreUpdateChecker.joomlaCurrentVersion,
```
```suggestion
            'joomla-current-version': PreUpdateChecker.joomlaCurrentVersion,
```

zero-24 commented on media/com_joomlaupdate/js/default.js:
```diff
@@ -71,19 +76,20 @@ function extractionMethodHandler(target, prefix)
         $.getJSON(PreUpdateChecker.config.serverUrl, {
             'joomla-target-version': PreUpdateChecker.joomlaTargetVersion,
+			'joomla-current-version': PreUpdateChecker.joomlaCurrentVersion,
+			'extension-version': $extension.data('extension-current-version'),
```
```suggestion
            'extension-version': $extension.data('extension-current-version'),
```

zero-24 commented on media/com_joomlaupdate/js/default.js:
```diff
@@ -71,19 +76,20 @@ function extractionMethodHandler(target, prefix)
-            extension.compatibleVersion = response.data.compatibleVersion;
-            extension.currentVersion = $extension.data('extensionCurrentVersion')
+            extension.serverError = 0;
+			extension.compatibilityData = response.data;
```
```suggestion
            extension.compatibilityData = response.data;
```

zero-24 reviewed (changes requested):
Some sugestions and CS changes will send you a PR for the PHP file.

zero-24 committed:
implement php cs changes

zero-24 commented:
Here is the PR for the suggested changes: https://github.com/GeraintEdwards/joomla-cms/pull/1 this already includes the change to the joomla loader and the removal of the additional notice message.

GeraintEdwards committed:
Merge pull request #1 from zero-24/310checker

php cs for the pre upgrade checker

Geraint2 committed:
Changes suggested by zero-24
* change updating icon to Joomla ajax-loader.gif
* Styling the results with margin left and right
Also corrected language type Jomla instead of Joomla

Geraint2 committed:
Revert accidentally committed file

Geraint2 committed:
More changes suggested by zero-24
* Remove double spacing after full stop in language strings

Geraint2 committed:
Changes suggested by zero-24
'is not supported' - awaiting feedback on capitalisation

Geraint2 committed:
Changes suggested by zero-24
* js file spacing

Geraint2 committed:
Changes suggested by zero-24
* Hathor template override php layout

GeraintEdwards commented:
@zero-24 
Hi Tobias
I've made almost all your suggested changes - I've left the heading capitalisation awaiting input from Brian (@brianteeman)
thanks
Geraint

Geraint2 committed:
Changes suggested by zero-24
* Add link to Joomla updater instructions for developer's benefit

zero-24 commented:
That looks great! Thanks @GeraintEdwards (and sorry that it took a week to come back to this)

I would be great to get another final test here maybe @alikon @adj9 @bayareajenn @toivo can give the latest code changes another try?

As for the headings it would be great to get some feedback from @brianteeman but this can also be changed when required in a later PR.

brianteeman commented:
Will check this morning - thanks for the reminder

brianteeman commented:
1. Sorry but I have no idea at all what these two columns are showing me

Upgrade Compatible Version | Currently Compatible Version

2.  Joomla Update System is Not Supported

Why does this text contain a link to how to setup an update server? That is not relevant to a user trying to update their site
> The extension developer is not currently offering any information about extension compatibility with different versions of Joomla. 

3. Extensions Requiring Upgrades to be Compatible
Does that mean I should update them now before I update Joomla? Or that I will have to make a note of them and update them after I have done the update


4. These are hard to understand and not consistent in language or tone with each other

> Extensions Probably Compatible
> Extensions Requiring Upgrades to be Compatible
> Update Server Offers No Compatible Version
> Joomla Update System is Not Supported

5. Update or Upgrade
Are we doing an update or an upgrade? Both words are used interchangeably - we should only use one

bayareajenn commented:
> Why does this text contain a link to how to setup an update server? That is not relevant to a user trying to update their site

Before there was a doc page dedicated to the pre-update check, I wanted a link to go somewhere so that extension devs could properly tag their extensions. This is no longer a good idea as the information can be obtained from docs.joomla.org/Pre-Update_Check.  @GeraintEdwards will you please link to /Pre-Update_Check.

When it comes to update vs. upgrade, the component was named Pre-Update Check before I ever saw it. The "upgrade/update" vs. "migration" was a huge discussion in the "update working group" during the 2.5 to 3.x migration time (2014). People wanted to minimize the mini-migration by using "update." I was against this, thus why the docs say "mini-migration" and not "update" or "upgrade." This is a mini-migration. So personally, I'd rather the entire component be called "pre-migration check." But I've given up on caring about that.

@GeraintEdwards I've already expressed my thoughts on wording changes and colours a number of times. I'll paste below what I think again.

> Thoughts:
> 
>     1. The second red bar I think should be gray - same as the missing compatibility tag is in the current version of pre-update check since a number of them will migrate on the one click to 4. To me red means NO and some of these are definitely YES.
> 
>     2. The titles in the bars I think should be clearer and ditch the description below. Suggestions:
> 
> 
> Green - Yes, tagged as compatible with Joomla 4.
> Yellow/Orange - May need an update to be compatible. Update or check with developer.
> Red - No compatiblity with Joomla 4. Check with developer on migration path.
> Gray/Second Red - Missing compatibility tag. Check with developer on migration path.
> 
>     1. We can put the deeper descriptions and more explanation in the doc (docs.joomla.org/Pre-Update_Check). Note that is a work in progress at this moment.
> 
>     2. Add to the Notice at the bottom:
> 
> 
> Pre-Update Check Documentation (link to docs.joomla.org/Pre-Update_Check)
> Dev Resources: Deploying an Update Server (https://docs.joomla.org/Deploying_an_Update_Server), Joomla Update System Requirement (https://extensions.joomla.org/support/knowledgebase/submission-requirements/joomla-update-system-requirement/), and Extension Manager Update (https://docs.joomla.org/Help39:Extensions_Extension_Manager_Update)

Those have been my suggestions from the beginning of testing this. Whatever y'all decide I'll go with. I'll just try and make it make sense in the docs after y'all decide what you want it to say.

GeraintEdwards commented:
>     1. Sorry but I have no idea at all what these two columns are showing me
> 
> Upgrade Compatible Version | Currently Compatible Version
The columns show versions of the extensions that are compatible with the Joomla version you want to upgrade to and versions that are compatible with your current version of Joomla (since your currently installed version of an extension may or may not be the most up to date).

Not easy to explain in a 2-3 word heading/
> 
>     1. Joomla Update System is Not Supported
>> 
> Why does this text contain a link to how to setup an update server? That is not relevant to a user trying to update their site

I agree with @bayareajenn that a link to the documentation page would be best overall.  I think Tobias' idea was to provider a pointer for developers here.
 
> > The extension developer is not currently offering any information about extension compatibility with different versions of Joomla.
> 
>     1. Extensions Requiring Upgrades to be Compatible
>        Does that mean I should update them now before I update Joomla? Or that I will have to make a note of them and update them after I have done the update
> 
The former - if you don't upgrade your newly upgraded version of Joomla may not work at all.  Many types of plugins could cause the upgraded site to be inaccessible or leave you unable to login.

>     2. These are hard to understand and not consistent in language or tone with each other
>  
> > Extensions Probably Compatible
> > Extensions Requiring Upgrades to be Compatible
> > Update Server Offers No Compatible Version
> > Joomla Update System is Not Supported

They are descriptions of the different combination of situations you could have.  The last 2 are basically 2 different types of "We don't know if there is a compatible version" but I think we need to maintain a distinction.

We also can't make 'bold' statements like 'Extensions Compatible' without a caveat of some sort.

I was hoping that the 'clarification wording under each heading would make their meaning a bit more clear.  

3rd heading could express something more like 'No Compatible Version Currently Available' and explain about update server underneath

4th could then read 'No Compatibility Information Available' - again with explanation about not supporting Joomla update server.

@bayareajenn 

>Green - Yes, tagged as compatible with Joomla 4.
>Yellow/Orange - May need an update to be compatible. Update or check with developer.

It quite possible that upgrading Joomla without upgrading the extension could lock you out of your site - so I think 'may need and update' is a bit too weak.

>Red - No compatiblity with Joomla 4. Check with developer on migration path.
>Gray/Second Red - Missing compatibility tag. Check with developer on migration path.

As a developer when I first saw 'missing compatibility tag' I had no idea what that meant and was what got me started on these tweaks in the first place.  I think its too technical a phrase for most people.

Again is 'migration path' a bit technical and and also the use of the word 'migration' may scare people with Joomla 1.5 experience.  I think we use the explanatory 'sub-heading' to elaborate on what the user should do and point them to the documentation.

Any suggestions?
> 
>     1. Update or Upgrade
>        Are we doing an update or an upgrade? Both words are used interchangeably - we should only use one
A quick scan of core language files shows update is used about 90% of the time and upgrade 10%

I'm happy to change these words - personally I prefer upgrade to update but its easier to go with the existing flow ;)'

brianteeman commented:
**Think like a user not a developer. Why am I on this page? To update joomla.** 

This check should provide me with anything I should know before I do the update. Version numbers are meaningless to me. The more complicated this is the more unlikely that people will either
1. not read any of it 
2. be too scared to do the update

So just because the site has to find all this information doesn't mean I need to see it. I would simply present basic information as in the screenshot below.

![image](https://user-images.githubusercontent.com/1296369/87326280-6d396e00-c52a-11ea-9a21-8fcddbb40866.png)

GeraintEdwards commented:
> **Think like a user not a developer. Why am I on this page? To update joomla.**

Its not just developers that like to have as much information as possible  ;)  

But I also think its useful for developers to see the version information in this context to flag where there are issues with update servers.  

What about a 'show more' type link to reveal more detailed information via a css slider?

> This check should provide me with anything I should know before I do the update. Version numbers are meaningless to me. The more complicated this is the more unlikely that people will either
> 
>     1. not read any of it
> 
>     2. be too scared to do the update
> 
> 
> So just because the site has to find all this information doesn't mean I need to see it. I would simply present basic information as in the screenshot below. Or even add the link to the developer site
> 
In general I agree but there are a couple of catches - for example:
1. Some extensions need to be upgraded before you upgrade to J4
2. Some extensions  need to be upgraded after you upgrade to J4 since they are not compatible with J3.x

We could extract this distinction from the data in my 2 columns and present it like you suggest with a 4th option 'Upgrade Required After Upgrading Joomla'

Links to extension developer sites and update details would be feasible and a good idea.

I think its worth flagging to users where they have extensions  which don't support the update server - users should be 'discouraged' from using these extensions so putting them in the 'sin bin' here is a good chance to prompt people to do a cleanout?

Finally I think we do need to make sure that users understand we are not guaranteeing compatibility of extensions - so a suitable disclaimer is important

brianteeman commented:
> We could extract this distinction from the data in my 2 columns and present it like you suggest with a 4th option 'Upgrade Required After Upgrading Joomla'

It's not needed. As soon as they complete the jooml update they will get that information

> I think its worth flagging to users where they have extensions which don't support the update server 

They are flagged in the Information unavailable section 

> Finally I think we do need to make sure that users understand we are not guaranteeing compatibility of extensions - so a suitable disclaimer is important

The disclaimer is already on the page
![image](https://user-images.githubusercontent.com/1296369/87328901-1cc40f80-c52e-11ea-89bd-278e805f05e3.png)

bayareajenn commented:
The migration docs will tell them that the pre-update check is an overview only. They need to check and assess each extension before hitting the button to 4.x.

GeraintEdwards commented:
@brianteeman 

> Some extensions need to be upgraded after you upgrade to J4 since they are not compatible with J3.x

I should have worded this differently.  Lets take an extension at v1.1 in J3 and v2.0 is released for J4 but v2.0 is not compatible with J3 and they MUST upgrade the extension after upgrading Joomla.   The user needs to be told an update is available so they know can upgrade Joomla but we need to make the distinction that they will need to upgrade the extension later???

It may be an academic point since most developers will have a 'transition' version of their extension but I don't know this for certain.

brianteeman commented:
As soon as they update joomla it will trigger the extension update check and they will be notified then

GeraintEdwards commented:
@brianteeman 

> As soon as they update joomla it will trigger the extension update check and they will be notified then

True, but it would be listed under the 'update required' heading and could lead people to spend time trying to perform an upgrade from J3.x that they cannot do leading to support questions and confusion.

Its not going to happen very often - if I were to guess maybe 5% of sites so most people wouldn't even see the output block since empty result blocks are not shown.

I'll work up an alternative output taking on board your ideas then we can compare and contrast?

Geraint2 committed:
Revised version to reflect some of Brian's ideas

GeraintEdwards commented:
@brianteeman 
Can you take a look at the revised version?  I know it will fail on code styling - need to get people's views of this version compared to the earlier version before I polish up the code.
thanks

GeraintEdwards committed:
Merge branch '3.10-dev' into compatibilitycheck

GeraintEdwards commented:
Screenshots so you get a quick impression

![Basic Output](https://user-images.githubusercontent.com/839810/87439385-a20bfa80-c5e8-11ea-9ca1-66c53f05e3e4.png)

![first section expanded](https://user-images.githubusercontent.com/839810/87439402-a7694500-c5e8-11ea-832d-2d3bf380ccff.png)

![second section expanded](https://user-images.githubusercontent.com/839810/87439413-ac2df900-c5e8-11ea-91ff-376593e98616.png)

![third section expanded](https://user-images.githubusercontent.com/839810/87439434-b18b4380-c5e8-11ea-86ed-a16bebf088f6.png)

brianteeman commented:
I'm a bit confused (as always) why the three different screenshots

ok worked it out - pressing the + reveals the version numbers

Not sure about that from a UI perspective as thats not a design pattern we use anywhere else and it doesnt solve the problem of understanding what the columns are.

For me the 2 columns are more than enough

My 2'6

GeraintEdwards commented:
@bayareajenn @zero-24 @toivo @alikon 
What does everyone else think of the merits of the 2 approaches?

infograf768 commented:
Imho, versions information should always be displayed. Headers can be modified if confusing.

softforge commented:
I think this has everything now. It is concise for those who are put off by the detail, it has the details so you can find the problem and know what to solve. It also helps the developers who are building their packages to see if and why they fail.

> Not sure about that from a UI perspective as that's not a design pattern we use anywhere else and it doesn't solve the problem of understanding what the columns are.

This is a one-off component that is needed as a migration tool. Its not a design pattern we often repeat

ReLater commented:
From my point of view: "No updates required" should be below all other lists. I'm not more interested in things that I don't have to do than in things I have to do. If "No updates required" becomes a long list you have to scroll down a lot before you see the imp0ortant entries.

richard67 commented:
> From my point of view: "No updates required" should be below all other lists. I'm not more interested in things that I don't have to do than in things I have to do. If "No updates required" becomes a long list you have to scroll down a lot before you see the imp0ortant entries.

I agree.

GeraintEdwards commented:
Rather than tweaking the code here is a screenshot of the results reordered and also with a short explanation and link to the docs as suggested by @bayareajenn 

![Joomla Update - with docs link and reordered](https://user-images.githubusercontent.com/839810/87632485-6761a980-c731-11ea-9095-7e7513e10248.png)

GeraintEdwards committed:
Merge branch '3.10-dev' into compatibilitycheck

Geraint2 committed:
Revised layout in traffic light colour sequence, using 'More Detail' and chevrons to indicate more detail available link and including introductory explanation

Geraint2 committed:
Revised layout in traffic light colour sequence, using 'More Detail' and chevrons to indicate more detail available link and including introductory explanation

Geraint2 committed:
Merge branch 'compatibilitycheck' of https://github.com/GeraintEdwards/joomla-cms into compatibilitycheck

Geraint2 committed:
Revised layout in traffic light colour sequence, using 'More Detail' and chevrons to indicate more detail available link and including introductory explanation

richard67 commented on administrator/language/en-GB/en-GB.com_joomlaupdate.ini:
```diff
@@ -58,8 +59,8 @@ COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_REQUIRING_UPDATES_TO_BE_COMPATIBLE="Upd
 COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_UPDATE_SERVER_OFFERS_NO_COMPATIBLE_VERSION_NOTES="Extension does not use the Joomla update system or the developer has provided no compatibility information."
-COM_JOOMLAUPDATE_VIEW_DEFAULT_SHOW_LESS_EXTENSION_COMPATIBILITY_INFORMATION="[ - ]"
-COM_JOOMLAUPDATE_VIEW_DEFAULT_SHOW_MORE_EXTENSION_COMPATIBILITY_INFORMATION="[ + ]"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_SHOW_LESS_EXTENSION_COMPATIBILITY_INFORMATION="[ Less Detail %s ]"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_SHOW_MORE_EXTENSION_COMPATIBILITY_INFORMATION="[ More Detail %s ]"
```
```suggestion
COM_JOOMLAUPDATE_VIEW_DEFAULT_SHOW_LESS_EXTENSION_COMPATIBILITY_INFORMATION="[ Less Details %s ]"
COM_JOOMLAUPDATE_VIEW_DEFAULT_SHOW_MORE_EXTENSION_COMPATIBILITY_INFORMATION="[ More Details %s ]"
```
Always plural when using more or less for countable things.

richard67 replied:
@GeraintEdwards In case if I'm wrong with my above suggestion (what could be), just mark the conversation as resolved.

GeraintEdwards replied:
In english I would argue we can use 'More Detail' and 'Less Detail' here.  If you were talking to a student about her poor  exam result and she asked "what was wrong with my answers?" you could say "I needed more detail".

If we switch to the plural we we should use 'Fewer Details' not 'Less Details'

Don't know about other languages?  

Happy to go with singular or plural?

"More Information"/"Less Information" may be an alternaitve

richard67 replied:
Yes, with "information" only singular is possible. And you are right, with countable stuff it's "fewer" and not "less". I'd suggest we ask @brianteeman to check the 2 language strings and suggest corrections if necessary. Personally I like the "More Information"/"Less Information" alternative.

brianteeman replied:
I agree with @GeraintEdwards

GeraintEdwards replied:
An interesting discussion here  on the topic https://forum.wordreference.com/threads/detail-vs-details.2822385/

richard67 replied:
I see I'm never too old to learn something new 😄 . Thanks. You can mark it as resolved if you want.

GeraintEdwards commented:
Just committed what I hope will be the final version (pending more/less discussion).

![Screenshot_2020-07-16 Joomla Update - Joomla 3 10 Git - Administration(2)](https://user-images.githubusercontent.com/839810/87656112-f92cdf00-c750-11ea-83a9-c1e49b5db9e8.png)

![Screenshot_2020-07-16 Joomla Update - Joomla 3 10 Git - Administration(3)](https://user-images.githubusercontent.com/839810/87656124-fc27cf80-c750-11ea-8324-4b3cdf09f5b3.png)

brianteeman commented on administrator/language/en-GB/en-GB.com_joomlaupdate.ini:
```diff
@@ -43,17 +43,33 @@ COM_JOOMLAUPDATE_VIEW_COMPLETE_HEADING="Joomla Version Update Status"
 COM_JOOMLAUPDATE_VIEW_DEFAULT_DIRECTIVE="Directive"
 COM_JOOMLAUPDATE_VIEW_DEFAULT_DOWNLOAD_IN_PROGRESS="Downloading update file. Please wait ..."
 COM_JOOMLAUPDATE_VIEW_DEFAULT_FTP_DIRECTORY="FTP Root"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXPLANATION_AND_LINK_TO_DOCS="The pre-update check provides you with information about the readiness of your server, settings and installed extensions for the update.<br>You can find more information about this page and how to prepare for updating Joomla in the <a class='pre-update-docs' href='https://docs.joomla.org/Pre-Update_Check'>pre update documentation</a>."
```
```suggestion
COM_JOOMLAUPDATE_VIEW_DEFAULT_EXPLANATION_AND_LINK_TO_DOCS="The pre-update check provides you with information about the readiness of your server, settings and installed extensions for the update.<br>You can find more information about this page and how to prepare for updating Joomla in the <a class='pre-update-docs' href='https://docs.joomla.org/Pre-Update_Check'>pre-update documentation</a>."
```

hyphenated for consistency

brianteeman commented:
One tiny change and then its perfect for me

GeraintEdwards committed:
Update administrator/language/en-GB/en-GB.com_joomlaupdate.ini

Co-authored-by: Brian Teeman <brian@teeman.net>

brianteeman commented:
I have tested this item :white_check_mark: successfully on b3e691aebdb5d6b4f288aadef6aece6b0cbc356a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29524">issues.joomla.org/tracker/joomla-cms/29524</a>.</sub>

Quy commented on media/com_joomlaupdate/js/default.js:
```diff
@@ -49,7 +57,51 @@ function extractionMethodHandler(target, prefix)
+		// No point creating and loading a component stylesheet for 4 settings
+		$('.compatibilitytypes img').css('height', '20px');
+		$('.compatibilitytypes').css('display', 'none').css('margin-left', 0);
+		// The currently processing line should show until its finished
```
```suggestion
		// The currently processing line should show until it’s finished
```

Quy commented on media/com_joomlaupdate/js/default.js:
```diff
@@ -97,34 +150,83 @@ function extractionMethodHandler(target, prefix)
+
+        // Process Target Version Extension Compatibility
+        if (extensionData.serverError) {
+			// An error occured -> show unknown error note
```
```suggestion
			// An error occurred -> show unknown error note
```
Fix in other instances.

GeraintEdwards committed:
Update media/com_joomlaupdate/js/default.js

Co-authored-by: Quy <quy@fluxbb.org>

GeraintEdwards committed:
Update media/com_joomlaupdate/js/default.js

Co-authored-by: Quy <quy@fluxbb.org>

bayareajenn commented:
I don't seem to be able to test it via Patch Tester. But thanks for adding a link to the docs @GeraintEdwards . Note, it should be pre-update _check_ rather than pre-update documentation just to be consistent with the title of the doc but I don't care as long as there's a link where more can be explained if needed.

GeraintEdwards commented:
> I don't seem to be able to test it via Patch Tester. But thanks for adding a link to the docs @GeraintEdwards . Note, it should be pre-update _check_ rather than pre-update documentation just to be consistent with the title of the doc but I don't care as long as there's a link where more can be explained if needed.

So the link text would be 'pre-update check documentation' instead of just 'pre-update documentation' ?

bayareajenn commented:
> So the link text would be 'pre-update check documentation' instead of just 'pre-update documentation' ?

Yes, sir. The doc is called Pre-Update Check so I think those are the words that should link to https://docs.joomla.org/Pre-Update_Check.

Geraint2 committed:
Language string tweak as per bayareajenn's suggestion

GeraintEdwards commented:
> > So the link text would be 'pre-update check documentation' instead of just 'pre-update documentation' ?
> 
> Yes, sir. The doc is called Pre-Update Check so I think those are the words that should link to https://docs.joomla.org/Pre-Update_Check.
Your wish is my command :)

bayareajenn commented:
Thanks for all your work on this @GeraintEdwards

alikon commented:
I have tested this item :white_check_mark: successfully on b3e691aebdb5d6b4f288aadef6aece6b0cbc356a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29524">issues.joomla.org/tracker/joomla-cms/29524</a>.</sub>

zero-24 commented:
Thanks for all involved in here and a special thanks to @GeraintEdwards 👍 
Will merge this one here for now. Lets try to get some feedback from the wider community on this one too so we can improve this where needed.
